### PR TITLE
Menu perf simulations

### PR DESCRIPTION
**Pull request by patrick-amido**
A few changes in this PR.
To test it run it against **stacks-java** which has an out the box working backend connecting to a cosmos db in the cloud.

Please refer to the readMe because running the tests has slightly changed.

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -23,17 +23,48 @@ Blog post by Amir Gharai which was used to set up the framework -
...
+`mvn clean gatling:test -Denv=local`
+
+By default, this will run a load test against all simulations with 1 user with a ramp up duration of 1 second.
 
-By default, this will run a load test against the GET menu api with 1 user with a ramp up duration of 1 second.
 There are optional parameters that can be added to change the load test:
 
-- `-Dusers=X` - This allows you to set the number of users that will be simulated in the tests
-- `-Drampup=X` - This sets the amount of seconds the test will ramp the test load from 0 users to X users in the test
+- `-DrampUsers=X` - This allows you to set the number of users that will be simulated in the tests.
```
Comment (github-patrick) [inline ReadMe.md:32]: comment test @patrick-amido

**Comment by AmirAmido** [comment]
here is the comment

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+target/
+.idea/
+*.iml
+*.logs
```
[the remark this answers is not in the document]
Comment (patrick-amido) [reply .gitignore:4]: Added.

File: src/test/scala/com/amido/scenarios/menu/CreateMenuScenario.scala [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+object CreateMenuScenario {
+
+  val createMenu: ScenarioBuilder = scenario("CREATE menu SUCCESS scenario")
+    .feed(randomName)
+    .exec(CreateMenuRequest.createMenu)
+
+  val createMenuError: ScenarioBuilder = scenario("CREATE menu error BAD REQUEST scenario")
+    .exec(CreateMenuRequest.createMenuError)
+}
```
[the remark this answers is not in the document]
Comment (patrick-amido) [reply src/test/scala/com/amido/scenarios/menu/CreateMenuScenario.scala:16]: @AmirAmido all scenarios are now in a MenuScenarios scala object file.

File: src/test/scala/com/amido/utils/tearDownDeleteMenuItems.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    for id in "${menuIds[@]}"; do
+      deleteResponse=$(curl -X DELETE "${baseUri}/v1/menu/${id}" -H  "${acceptHeader}")
+      echo "${deleteResponse}"
+    done
+  fi
+
+  menuIds=()
+  count=$(curl -X GET "${url}" -H  "${acceptHeader}" | jq -r '.results | length')
+  echo "Number of menu items to delete from latest request : ${count}"
+done
```
[the remark this answers is not in the document]
Comment (patrick-amido) [reply src/test/scala/com/amido/utils/tearDownDeleteMenuItems.sh:41]: @AmirAmido  Yes, the `PERF TEST` prefix is a marker to easily identify any created resources written to a database "Menu" table that needs to be deleted. I will mention this in the readMe file.

File: src/test/scala/com/amido/requests/menu/CreateMenuRequest.scala [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    .headers(httpHeaders)
+    .check(status is 201)
+    .check(jsonPath("$.id").saveAs("menuId"))
+
+  val createMenuError: HttpRequestBuilder = http("Create Menu Error")
+    .post(baseUrl + MenuPath)
+    .body(RawFileBody(createMenuErrorResource.getPath())).asJson
+    .headers(httpHeaders)
+    .check(status is 400)
+}
```
[the remark this answers is not in the document]
Comment (patrick-amido) [reply src/test/scala/com/amido/requests/menu/CreateMenuRequest.scala:39]: @AmirAmido all the requests are now in 1 file.

File: src/test/resources/bodies/CreateMenuError.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "",
+  "description": "Invalid payload missing blank name",
+  "enabled": true,
+  "tenantId": "d290f1ee-6c54-4b01-90e6-d701748f0851"
+}
```
[the remark this answers is not in the document]
Comment (patrick-amido) [reply src/test/resources/bodies/CreateMenuError.json:6]: @AmirAmido error scenarios have been removed.

**Pull request merged**